airbyte-ci/connectors/regression-testing/src/regression_testing/main.py

```diff
@@ -0,0 +1,191 @@
+import asyncio
+import logging
+import sys
+from typing import Dict, Optional
+
+import click
+import dagger
+from airbyte_protocol.models import ConfiguredAirbyteCatalog
+
+from .backends import BaseBackend, FileBackend
+from .comparators import DiffComparator
+from .connector_runner import ConnectorRunner, SecretDict
+from .utils import get_connector, get_connector_config, get_state
+
+logging.basicConfig(level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+COMMANDS = ["check", "discover", "read", "read-with-state", "spec"]
+
+
+async def _main(
+    connector_name: str,
+    control_image_name: str,
+    target_image_name: str,
+    output_directory: str,
+    command: Optional[str],
+    config_path: Optional[str],
+    catalog_path: Optional[str],
+    state_path: Optional[str],
+):
+    if control_image_name in ("dev", "latest"):
+        control_image_name = f"airbyte/{connector_name}:{control_image_name}"
+
+    if target_image_name in ("dev", "latest"):
+        target_image_name = f"airbyte/{connector_name}:{target_image_name}"
+
+    # TODO: add backend options
+    if not output_directory:
+        raise NotImplementedError(f"An output directory is required; only file backends are currently supported.")
+
+    config = get_connector_config(config_path)
+    catalog = ConfiguredAirbyteCatalog.parse_file(catalog_path) if catalog_path else None
+    state = get_state(state_path) if state_path else None
+
+    async with (dagger.Connection(config=dagger.Config(log_output=sys.stderr)) as client):
+        control_connector = await get_connector(client, connector_name, control_image_name)
+        target_connector = await get_connector(client, connector_name, target_image_name)
+
+        # TODO: maybe use proxy to cache the response from the first round and use the cache for the second round
+        #   (this may only make sense for syncs with an input state)
+        if command == "all":
```

Comment:
can we instead change the main signature to take a `commands` list and iterate on commands? It will help you avoid the `if/else`

Reply:
Yep! Done.